samcli/yamlhelper.py

```diff
@@ -46,22 +47,34 @@ def intrinsics_multi_constructor(loader, tag_prefix, node):
     return {cfntag: value}
 
 
+def _dict_representer(dumper, data):
+    return dumper.represent_dict(data.items())
+
+
 def yaml_dump(dict_to_dump):
     """
     Dumps the dictionary as a YAML document
     :param dict_to_dump:
     :return:
     """
+    yaml.SafeDumper.add_representer(OrderedDict, _dict_representer)
     return yaml.safe_dump(dict_to_dump, default_flow_style=False)
 
 
+def _dict_constructor(loader, node):
+    # Necessary in order to make yaml merge tags work
```

Comment:
According to [CloudFormation template formats](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-formats.html), aliases and hash merges are not supported, but I think it's still useful to be able to parse the syntax.

Reply:
We cannot support this at the moment. The package command (and a similar yamlhelper) lives within AWS CLI. Supporting this here means, customers can write this in template, test locally, but fail during deployments to AWS CloudFormation.

Reply:
I have a similar [PR](https://github.com/aws/aws-cli/pull/3841) opened to aws-cli. The original yamlhelper is able to parse the merge syntax, and the `package` command could run without any issues, but the template would fail `cloudformation validate-template`. 

`loader.flatten_mapping(node)` is only needed for using customized dict constructor.    I don't feel strongly to keep it, but I want to make sure I don't break any existing functionalities.

Reply:
I see. This is to have support for what was perviously there. The initial comment made it sound like you were proposing we add this support.